tpm/tpm_test.go

```diff
@@ -70,6 +70,61 @@ func TestGetManufacturer(t *testing.T) {
 	t.Logf("TPM VendorID: %v\n", vendorID)
 }
 
+func TestGetAlgs(t *testing.T) {
+	rwc := openTPMOrSkip(t)
+	defer rwc.Close()
+
+	algs, err := GetAlgs(rwc)
+	if err != nil {
+		t.Fatalf("Couldn't read Algorithms: %v\n", err)
+	}
+	for iterator, item := range algs {
```

Comment:
This looks wrong.
`GetAlgs` returns a list of Algorithms that are supported.
Here, the test code assumes that index of algs is the Algorithm identifier (it's not, it's just slice index) and the value is whether it's enabled (the value is Algorithm identifier).

You probably want something like:
```
want := []Algorithm{AlgRSA, AlgSHA, AlgHMAC, AlgMGF1}
outer:
for _, alg := range want {
  for _, got := range algs {
    if got == alg {
      continue outer
    }
    t.Errorf("GetAlgs returned %v, which does not contain a required algorithm %v", algs, alg)
  }
}
```

Reply:
Alright. I see my mistake.
Thank you :)